apps/documentation/pages/apps/graphql/index.md

```diff
@@ -0,0 +1,107 @@
+
+The code is organised by feature, as you can see in `apps/graphql/src/apps`.
+
+```
+apps/graphql/
+├── index.js
+├── src
+│   ├── Schema.js                           # lists all the queries and mutations
+│   ├── apps
+│   │   ├── booking
+│   │   │   ├── Booking.js                  # Flow types concerning booking feature
+│   │   │   ├── __datasets__                # datasets to mock API responses
+│   │   │   │   ├── AllBookings.json
+│   │   │   │   ├── booking-16463447.json
+│   │   │   │   ├── booking-6676524.json
+│   │   │   │   └── booking-8142828.json
+│   │   │   ├── dataloaders                 # contains the logic to fetch and sanitize data
+│   │   │   │   ├── BookingDetail.js
+│   │   │   │   └── Bookings.js
+│   │   │   ├── queries                     # GraphQL queries concerning booking feature
+│   │   │   │   ├── BookingDetail.js
+│   │   │   │   ├── CustomerBookings.js
+│   │   │   │   └── __tests__
+│   │   │   └── types                       # GraphQL types concerning booking feature
+│   │   │       ├── enums
+│   │   │       ├── inputs
+│   │   │       └── outputs
+│   │   ├── common
+│   │   │   └── types
+│   │   ├── itinerary
+│   │   │   ├── Itinerary.js
+│   │   │   ├── __datasets__
+│   │   │   ├── __mocks__
+│   │   │   ├── dataloaders
+│   │   │   ├── helpers
+│   │   │   ├── queries
+│   │   │   └── types
+│   │   └── location
+│   │       ├── Location.js
+│   │       ├── __datasets__
+│   │       ├── dataloaders
+│   │       ├── queries
+│   │       └── types
+│   ├── server.js
+│   └── services                            # lists of services/helpers useful across all features
+│       ├── fetch
+│       ├── graphqlContext
+│       ├── logger
+│       └── testingTools
+```
+
+Each feature has a similar folder structure:
+
+```
+├── feature
+│    ├── Feature.js             # Flow types concerning feature
+│    ├── __datasets__           # datasets to mock API responses
+│    ├── dataloaders            # contains the logic to fetch and sanitize data
+│    ├── queries                # GraphQL queries concerning feature
+│    ├── mutations              # GraphQL mutations concerning feature
+│    ├── resolvers              # more complex GraphQL resolvers
+│    └── types                  # GraphQL types concerning feature
+│        ├── enums
+│        ├── inputs             # definitions of INPUT types exposed to the client, e.g. arguments in a query
+│        └── outputs            # definitions of OUTPUT types exposed to the client, e.g. GraphQL type for GPS coordinates
+
+```
+
+The `dataloaders` folder is where the logic to connect to external APIs is stored. It has this name because it relies under the hood on https://github.com/facebook/dataloader which allows batching (i.e. trigger one API request even if your code needs to call the same API endpoint with different queries) and caching (i.e. using previously obtained data and not calling the API endpoint if the same call has already been made before).
```

Comment:
Maybe we should mention that it is pr-request cache?

Reply:
Sure, I will add that with link to corresponding line of code.

Reply:
🚀